### DISCUSSION TITLE
Does `git-stack` function with worktrees?

### DISCUSSION THREAD

**Discussion by dcunited001** [gitext-rs/git-stack#335] [General]
I can't figure out how to get any branches to be unprotected. I was able to manually move it forward using `git pull --rebase`

I think the key is the output `no remote` which results from the conditional `is_protected(...)` but I don't know `rust`.

+ I have another checkout without worktrees where things seemed to go more smoothly, but I can't really tell, as I haven't see `git-stack` in action. 
+ I've tried testing changing/removing remotes from `.bare/config` as well as changing/removing merge
+ I have two remotes and I've tried variously changing those as well

### Output

```log
A Merge PR#0
├─┐
│ ⌽ B  
│ ⌽ C  
│ ⌽ D 
│ ⌽ E (no remote)  
│
⌽ PR2
⌽ main 
Stack from other users: PR2
```

### Worktree file structure

My worktree setup looks like this:

```text
.
├── .bare
│   ├── branches
│   ├── config
│   ├── HEAD
│   ├── refs
│   └── worktrees
│       └── main
│           └── gitdir
├── .git
├── .bin
└── main
    ├── .git
    ├── .github
    ├── .gitignore
    ├── LICENSE
    ├── README.md
    └── site
```

### Git Log

`A` is the upstream (formerly main) and `N` is the new base

```gitlog
* E'
|   (feat-local-co-manual-rebase)
* D'
|
* C'
|
* B'
|
*   N   Merge PR#1
|\    (origin/main, main)
| * M
| |   (origin/PR-2, PR-2)
| * L
|/
*   K
|\
| * J
| |   (origin/PR-1, PR-1)
| * I
|/
| * H
| |   (dc/feat-F2)
| * G
| |
| | * F
| |/    (dc/feat-F1)
| * E
| |   (HEAD -> feat-local-co-2, origin/feat)
| * D
| |
| * C
| |
| * B
|/
* A
```

I'm kinda adopting worktrees (with a bare checkout) at the same time as adopting `git-stack`. I thought I had checked this repo for compatibility with worktrees beforehand, since I didn't want to introduce too many variables simultaneously. The rubber ducky is starting to indicate that my problems might be caused by worktrees. i'm not too sure how to manage my editor configs and ignored files ... not without file-system traversal & linking andways

I'm somewhat new to the idea of stacks. Though not, since it's how I would've wanted to work, I've thought about it alot.

**Comment by epage** [comment]
We are tracking worktrees at #23.  For myself, I almost never use them and there has been little interest in them so far.  If you'd like to contribute support for it, you are welcome to.

**Reply by dcunited001** [reply]
Thanks for the reply. I'll take a closer look, but `git-stack` seems to work well outside of my worktrees clone. The cause is probably configuration/etc and will be more apparent once I get more use with the tool.

I had been very excited about worktrees, but now I'm not so sure.

**Reply by dcunited001** [reply]
I believe I had skipped the `git config --add core.logallrefupdates true` part mentioned in this guide [Sliced bread: git-worktree and bare repo](https://blog.cryptomilk.org/2023/02/10/sliced-bread-git-worktree-and-bare-repo/)

I've only tried read commands, no write commands, but the `git-stash` output seems to be correct:

```log
cfcfa5d Way back at trying a different theme
├─┐
│ ⌽ f5757e2
│ ⌽ cherrypick-decomposition (ready) Revert "squashed merges"
│
⌽ 42edda1 "Squashed Merges"
├─┐
│ ⌽ 782f35f Squashed commit of the following:
│ ⌽ 7e979ac Adding post for instagram
│ ⌽ 8b4e2ff attempting a fix for instagram embed
│ ⌽ c41e2d8 
│ ⌽ 3aab999 Squashed commit of the following:
│ ⌽ 2f85cf3 Squashed commit of the following:
│ ⌽ f50dd0d Squashed commit of the following:
│ ⌽ 85d0fd4 Squashed commit of the following:
│ ⌽ 0abf518 Squashed commit of the following:
│ ⌽ 89ef31a 
│ ⌽ 57ad7de Merge pull request #10
│ ⌽ 49b060a Squashed commit of the following:
│ ⌽ 93786d0 Merge pull request #11
│ ⌽ cherrypick (ready), untheme (ready) Merge pull request #12
│
⌽ main Merge pull request #12 
Empty stacks: tools
Stack from other users: feature-abc1, feature abc2
```

I'm trying to produce a specific changeset above. My git ettiquette isn't great and I don't think i'll try to merge the above, but i've thought alot about cherrypicking and splitting revert commits to squash ... so the actual caveats to doing so interests me. It's a bit like the `ABA^T^` decomposition from linear algebra. 

... Anyways, it also gives me enough data to know whether the `git stack` seems to be working.
